chromium_src/components/bookmarks/browser/bookmark_model.cc

```diff
@@ -7,20 +7,22 @@
 
 namespace bookmarks {
 
-// Move bookmarks under "Other Bookmarks" permanent node to a same name folder
-// at the end of "Bookmark Bar" permanent node
-void BraveMigrateOtherNode(BookmarkModel* model) {
+// Move bookmarks under "Other Bookmarks" folder from
+// https://github.com/brave/brave-core/pull/3620 to original permanent node
+void BraveMigrateOtherNodeFolder(BookmarkModel* model) {
   DCHECK(model);
-  if (!model->other_node()->children().empty()) {
-    const bookmarks::BookmarkNode* new_other_node =
-        model->AddFolder(model->bookmark_bar_node(),
-                         model->bookmark_bar_node()->children().size(),
-                         model->other_node()->GetTitledUrlNodeTitle());
-    size_t children_size = model->other_node()->children().size();
+  if (!model->bookmark_bar_node()->children().size())
+    return;
+  const bookmarks::BookmarkNode* possible_other_node_folder =
+    model->bookmark_bar_node()->children().back().get();
+  if (possible_other_node_folder->GetTitledUrlNodeTitle() ==
```

Comment:
Why we have idea possible_other_node_folder is at the end of bookmarks bar?

Reply:
This is for the result of previous migration of "removing other bookmarks". We are not sure if it is still there, if it is, we migrate it back.

Reply:
oh, because we had put  `Other bookmarks` into the end position on `Bookmarks bar`.
So if we cannot find  `Other bookmarks` as the last bookmark item in the end of `Bookmarks bar`, like for example when some site was bookmarked at the position  after  `Other bookmarks`, then we will not try to iterate and find anything looks like `Other bookmarks` folder. - seems good because it is impossible to find the exact old created `Other bookmarks` instance.

Here I have just a paranoiac question, what if the last item will have a name `Other bookmark` , but accidentally will not be a folder and just a regular bookmark? Maybe it worth to add `if (possible_other_node_folder->is_folder() && possible_other_node_folder->GetTitledUrlNodeTitle() == ...` ?

Reply:
it seems nothing bad should happened, because `const TreeNodes& children() const { return children_; }` it will be just an empty vector. So I don't insist on this change.

Reply:
`BraveMigrateOtherNodeFolder` processes the case when current PR is applied on profile from the  old version of Brave without 1st migration of `Other bookmarks` - good.

Reply:
I will add the check and also add the case to test, thanks

Reply:
fixed